src/main/java/Duke.java

```diff
@@ -25,44 +25,70 @@ public static void main(String[] args) {
                   }
             }else{ 
                 String inputnoun = line.split(" ")[0];
-                String taskName = line.split(" ", 2)[1];
-                if(inputnoun.equals("done")){
-                    for (int i = 0; i < listOfStuff.size(); i++) {
-                        Integer indexOfTask = i + 1;
-                        String stringFromUser = line.replaceAll("\\D+","");
-                        Integer indexFromUser = Integer.parseInt(stringFromUser);
-                        if(indexOfTask == indexFromUser ){
-                            Task currentTask = listOfStuff.get(i);
-                            currentTask.setStatus(true);
-                            String message = "Nice! I've marked this task as done:\n" + currentTask;
-                            System.out.println(message);
+
+                try{
+                    if((!inputnoun.equals("done")) &&
```

Comment:
Perhaps it might be better to have some separation of concerns here. For each different value for the `inputnoun`, maybe you can consider calling a different function, so that the functions encapsulate the logic required for each `inputnoun` value.